tap_github/organization_streams.py

```diff
@@ -0,0 +1,168 @@
+        """
+        Override the parent method to allow skipping API calls
+        if the stream is deselected and skip_parent_streams is True in config.
+        This allows running the tap with fewer API calls and preserving
+        quota when only syncing a child stream. Without this,
+        the API call is sent but data is discarded.
+        """
+        if (
+            not self.selected
+            and "skip_parent_streams" in self.config
+            and self.config["skip_parent_streams"]
+            and context is not None
+        ):
+            # build a minimal mock record so that self._sync_records
+            # can proceed with child streams
+            yield {
+                "org": context["org"],
+            }
+        else:
+            yield from super().get_records(context)
+
+    schema = th.PropertiesList(
+        th.Property("login", th.StringType),
+        th.Property("id", th.IntegerType),
+        th.Property("node_id", th.StringType),
+        th.Property("url", th.StringType),
+        th.Property("repos_url", th.StringType),
+        th.Property("events_url", th.StringType),
+        th.Property("hooks_url", th.StringType),
+        th.Property("issues_url", th.StringType),
+        th.Property("members_url", th.StringType),
+        th.Property("public_members_url", th.StringType),
+        th.Property("avatar_url", th.StringType),
+        th.Property("description", th.StringType),
+    ).to_dict()
+
+
+class TeamsStream(GitHubRestStream):
+    name = "teams"
+    primary_keys = ["id"]
+    path = "/orgs/{org}/teams"
+    ignore_parent_replication_key = True
+    parent_stream_type = OrganizationStream
+    state_partitioning_keys = ["org"]
+
+    def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
+        new_context = {"team_slug": record["slug"]}
+        if context:
+            return {
+                **context,
+                **new_context,
+            }
+        return new_context
+
+    schema = th.PropertiesList(
+        # Parent Keys
+        th.Property("org", th.StringType),
+        # Rest
+        th.Property("id", th.IntegerType),
+        th.Property("node_id", th.StringType),
+        th.Property("url", th.StringType),
+        th.Property("html_url", th.StringType),
+        th.Property("name", th.StringType),
+        th.Property("slug", th.StringType),
+        th.Property("description", th.StringType),
+        th.Property("privacy", th.StringType),
+        th.Property("permission", th.StringType),
+        th.Property("members_url", th.StringType),
+        th.Property("repositories_url", th.StringType),
+        th.Property("parent", th.StringType),
+    ).to_dict()
+
+
+class TeamMembersStream(GitHubRestStream):
+    name = "team_members"
+    primary_keys = ["id"]
+    path = "/orgs/{org}/teams/{team_slug}/members"
+    ignore_parent_replication_key = True
+    parent_stream_type = TeamsStream
+    state_partitioning_keys = ["org", "team_slug"]
```

Comment:
With [this PR](https://github.com/MeltanoLabs/tap-github/pull/76) we've done the route of ordering keys by decreasing expected cardinality.
```suggestion
    state_partitioning_keys = ["team_slug", "org"]
```
> columns with highest cardinality (most distinct values) should come first, followed by lower cardinality.